### PR TITLE
feat: Inject multiple connections with polaris-nest

### PR DESCRIPTION
**Pull request by ronkatz96**

### CODE REVIEW

File: packages/polaris-test/shared-resources/connection-options.ts [hunk 1]
```diff
@@ -11,3 +11,15 @@ export const connectionOptions: ConnectionOptions = {
     schema: process.env.SCHEMA_NAME,
     extra: { max: 10 },
 };
+
+export const connectionOptionsRon: ConnectionOptions = {
```
Comment (itaykl) [inline packages/polaris-test/shared-resources/connection-options.ts]: https://i.pinimg.com/736x/ee/a5/e6/eea5e6604159a6405c832710cd7319dc.jpg

File: packages/polaris-nest/guides/README.md [hunk 2]
```diff
@@ -44,20 +44,34 @@ export class AuthorModule {}
 ```typescript
 @Injectable({ scope: Scope.REQUEST })
 export class AuthorService {
+    private authorRepository: PolarisRepository<Author>;
     constructor(
-        @InjectRepository(Author)
-        private readonly authorRepository: PolarisRepository<Author>,
-        @InjectConnection()
-        private readonly connection: PolarisConnection,
         @Inject(CONTEXT) private readonly ctx: TestContext,
```
Comment (chenshoo) [inline packages/polaris-nest/guides/README.md]: remove         `@Inject(CONTEXT) private readonly ctx: TestContext,`

File: packages/polaris-nest/guides/README.md [hunk 3]
```diff
@@ -44,20 +44,34 @@ export class AuthorModule {}
...
-    ) {}
+        @Inject(PolarisTypeORMInjector)
+        private readonly polarisTypeORMInjector: PolarisTypeORMInjector,
+    ) {
+        this.authorRepository = this.polarisTypeORMInjector.getRepository(Author);
+    }
 
     public async create(firstName: string, lastName: string): Promise<Author> {
         const author = new Author(firstName, lastName);
         return ((await this.authorRepository.save(this.ctx, author)) as unknown) as Promise<Author>;
```
Comment (chenshoo) [inline packages/polaris-nest/guides/README.md]: return ((await this.authorRepository.save(this.ctx, author)) as unknown) as Promise<Author>;
remove this.ctx

File: packages/polaris-test/nest-server-code-first/graphql/services/author.service.ts [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 import { TestContext } from '../../../shared-resources/context/test-context';
 import { Author } from '../../../shared-resources/entities/author';
 
 @Injectable({ scope: Scope.REQUEST })
 export class AuthorService {
     private authorRepository: PolarisRepository<Author>;
     constructor(
-        @InjectConnection()
-        connection: PolarisConnection,
         @Inject(CONTEXT) private readonly ctx: TestContext,
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-code-first/graphql/services/author.service.ts:12]: remove `@Inject(CONTEXT) private readonly ctx: TestContext,`

File: packages/polaris-test/nest-server-code-first/graphql/services/book.service.ts [hunk 5]
```diff
@@ -22,13 +20,13 @@ export class BookService {
     private bookRepository: PolarisRepository<Book>;
     private authorRepository: PolarisRepository<Author>;
     constructor(
-        @InjectConnection()
-        connection: PolarisConnection,
         @Inject(CONTEXT) private readonly ctx: TestContext,
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-code-first/graphql/services/book.service.ts:23]: remove `@Inject(CONTEXT) private readonly ctx: TestContext,`

File: packages/polaris-test/nest-server-schema-first/graphql.ts [hunk 6]
```diff
@@ -1,4 +1,3 @@
-
 /** ------------------------------------------------------
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-schema-first/graphql.ts]: run npm i

File: packages/polaris-test/nest-server-schema-first/graphql/services/author.service.ts [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
     private authorRepository: PolarisRepository<Author>;
     constructor(
-        @InjectConnection()
-        connection: PolarisConnection,
         @Inject(CONTEXT) private readonly ctx: TestContext,
+        @Inject(PolarisTypeORMInjector)
+        private readonly polarisTypeORMInjector: PolarisTypeORMInjector,
     ) {
-        this.authorRepository = connection.getRepository(Author, ctx);
+        this.authorRepository = this.polarisTypeORMInjector.getRepository(Author);
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-schema-first/graphql/services/author.service.ts:16]: remove `@Inject(CONTEXT) private readonly ctx: TestContext,`

File: packages/polaris-test/nest-server-schema-first/graphql/services/book.service.ts [hunk 8]
```diff
@@ -22,13 +20,13 @@ export class BookService {
     private bookRepository: PolarisRepository<Book>;
     private authorRepository: PolarisRepository<Author>;
     constructor(
-        @InjectConnection()
-        connection: PolarisConnection,
         @Inject(CONTEXT) private readonly ctx: TestContext,
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-schema-first/graphql/services/book.service.ts:23]: remove `@Inject(CONTEXT) private readonly ctx: TestContext,`

File: packages/polaris-test/nest-server-schema-first/graphql/services/chapter.service.ts [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
 @Injectable({ scope: Scope.REQUEST })
 export class ChapterService {
     private bookRepository: PolarisRepository<Book>;
     private chapterRepository: PolarisRepository<Chapter>;
     constructor(
-        @InjectConnection()
-        connection: PolarisConnection,
+        @Inject(PolarisTypeORMInjector)
+        private readonly polarisTypeORMInjector: PolarisTypeORMInjector,
         @Inject(CONTEXT) ctx: TestContext,
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-schema-first/graphql/services/chapter.service.ts:16]: remove `@Inject(CONTEXT) ctx: TestContext,`

File: packages/polaris-test/nest-server-schema-first/graphql/services/pen.service.ts [hunk 10]
```diff
@@ -1,22 +1,22 @@
...
 @Injectable({ scope: Scope.REQUEST })
 export class PenService {
     private authorRepository: PolarisRepository<Author>;
     private penRepository: PolarisRepository<Pen>;
     constructor(
-        @InjectConnection()
-        connection: PolarisConnection,
+        @Inject(PolarisTypeORMInjector)
+        private readonly polarisTypeORMInjector: PolarisTypeORMInjector,
         @Inject(CONTEXT) ctx: TestContext,
```
Comment (chenshoo) [inline packages/polaris-test/nest-server-schema-first/graphql/services/pen.service.ts:16]: remove `@Inject(CONTEXT) ctx: TestContext,`

**Review by chenshoo — APPROVED** [review]

**Pull request merged**